seatunnel-core/seatunnel-spark-starter/seatunnel-spark-starter-common/src/main/java/org/apache/seatunnel/core/starter/spark/execution/TransformExecuteProcessor.java

```diff
@@ -129,10 +130,11 @@ public List<DatasetTableInfo> execute(List<DatasetTableInfo> upstreamDataStreams
 
     private Dataset<Row> sparkTransform(SeaTunnelTransform transform, DatasetTableInfo tableInfo) {
         Dataset<Row> stream = tableInfo.getDataset();
-        SeaTunnelDataType<?> inputDataType = tableInfo.getCatalogTable().getSeaTunnelRowType();
+        SeaTunnelDataType<?> inputDataType =
+                tableInfo.getCatalogTables().get(0).getSeaTunnelRowType();
```

Comment:
Does Covenant `transform` only deal with the first one?

Reply:
This is because 'ransform' now supports only single tables